### PR TITLE
fix bug of keeping toc metadata

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

**Comment by OsmondJiang** [comment]
#3306

File: src/docfx/build/legacy/files/LegacyTableOfContents.cs [hunk 1]
```diff
@@ -74,6 +67,9 @@ private sealed class LegacyTableOfContentsMetadata
 
             [JsonProperty(PropertyName = "pdf_name")]
             public string PdfName { get; set; }
+
+            [JsonExtensionData]
+            public JObject Metadata { get; set; }
```
Comment (yufeih) [inline src/docfx/build/legacy/files/LegacyTableOfContents.cs]: Is `experimental` and `experimental_id` needed for non-legacy output ?

Comment (OsmondJiang) [reply src/docfx/build/legacy/files/LegacyTableOfContents.cs]: yes, it's used for AB testing

Comment (OsmondJiang) [reply src/docfx/build/legacy/files/LegacyTableOfContents.cs]: and we just pass through that, don't need to parse the metadata, it's wrote by writer

**Pull request merged**